### ISSUE TITLE
Fix port selection for ddp runs

### ISSUE THREAD

**Issue by Whisky-Jack** [WatChMaL/WatChMaL#6]
At the moment port is set manually to default value. Port selection should be done automatically, and without collision so that it is possible to run multiple ddp processes at once without having to manually set the value (see https://github.com/PyTorchLightning/pytorch-lightning/issues/234). Default value should be added as an option in the config.

**Comment by nickwp** [comment]
Can the port number be chosen simply based on which GPU(s) the process is using? We should never have multiple processes using the same GPU I think?

**Comment by Whisky-Jack** [comment]
Yeah probably. I could use a default port number, and add to it the base gpu. Will this cover/do we care about running on cedar?

**Comment by nickwp** [comment]
We do want to be able to use Cedar eventually, but I'm not sure how that will work. If this doesn't work for that case then probably we'll want an option using the SLURM job ID, similar to as suggested in the issue you linked to, but I think we can deal with that if / when it becomes an issue.

**Comment by nickwp** [comment]
Current status on this is that the port number is chosen by the first GPU number. This should be fine when running on either:
* triumf-ml1 or similar servers where you choose particular GPUs but can have multiple watchmal processes running simultaneously
* Compute Canada clusters where a job requests a whole node and all GPUs of that node are used.
It will not work on Compute Canada clusters if we happen to get two jobs allocated to exactly the same node simultaneously, but this should be rare and it's recommended to request whole nodes anyway, since watchmal can efficienctly use all GPUs with the appropriate batch size.
